servers/zms/src/main/java/com/yahoo/athenz/zms/store/impl/jdbc/JDBCConnection.java

```diff
@@ -3982,9 +3995,153 @@ void getAthenzDomainPolicies(String domainName, int domainId, AthenzDomain athen
             throw sqlError(ex, caller);
         }
 
+        // add services tags
+        addTagsToPolicies(policyMap, athenzDomain.getName());
+
         athenzDomain.getPolicies().addAll(policyMap.values());
     }
 
+    void addTagsToPolicies(Map<Integer, Policy> policyMap, String domainName) {
+
+        Map<String, Map<String, TagValueList>> domainPolicyTags = getDomainPolicyTags(domainName);
+        if (domainPolicyTags != null) {
+            for (Map.Entry<Integer, Policy> policyEntry : policyMap.entrySet()) {
+                Map<String, TagValueList> policyTag = domainPolicyTags.get(ZMSUtils.extractPolicyName(domainName, policyEntry.getValue().name) + ":" + policyEntry.getValue().getVersion());
+                if (policyTag != null) {
+                    policyEntry.getValue().setTags(policyTag);
+                }
+            }
+        }
+    }
+
+    Map<String, Map<String, TagValueList>> getDomainPolicyTags(String domainName) {
+        final String funcCaller = "getDomain Policy Tags";
+        Map<String, Map<String, TagValueList>> domainResourceTags = null;
+
+        try (PreparedStatement ps = con.prepareStatement(SQL_GET_DOMAIN_POLICY_TAGS)) {
```

Comment:
I think your Join query should ignore inactive policies.
In addition - please add multi-version policies tests to validate we are returning tags only for the active policy

Reply:
this query shouldn't ignore inactive policy version because in the end it uses for adding the tags for all the policies in a specific domain.